azure/components/db/setup.ftl

```diff
@@ -0,0 +1,145 @@
+[#ftl]
+
+[#macro azure_db_arm_generationcontract_solution occurrence]
+    [@addDefaultGenerationContract subsets=["prologue", "parameters", "template"] /]
+[/#macro]
+
+[#macro azure_db_arm_setup_solution occurrence]
+
+    [@debug message="Entering" context=occurrence enabled=true /]
+
+    [#local core       = occurrence.Core]
+    [#local solution   = occurrence.Configuration.Solution]
+    [#local resources  = occurrence.State.Resources]
+    [#local attributes = occurrence.State.Attributes]
+   
+    [#local engine = solution.Engine]
+    [#local engineVersion = solution.EngineMinorVersion?has_content?then(
+    solution.EngineVersion + solution.EngineMinorVersion, 
+    solution.EngineVersion)]
+
+    [#-- Resources --]
+    [#local server   = resources["dbserver"]]
+    [#local db       = resources["database"]]
+    [#local configs  = resources["dbconfigs"]]
+    [#local vnetRule = resources["dbvnetrule"]]
+
+    [#local masterAccount      = attributes["USERNAME"]]
+    [#local masterSecret       = attributes["SECRET"]]
+    [#local masterSecretId     = formatId(db.Id, SECRET_ATTRIBUTE_TYPE)]
+    [#local masterSecretLength = solution.GenerateCredentials.CharacterLength]
+
+    [#-- Hibernation --]
+    [#local hibernate  = solution.Hibernate.Enabled && isOccurrenceDeployed(occurrence)]
+    [#local createMode = "Default"]
+
+    [#-- Baseline Links --]
+    [#local baselineLinks      = getBaselineLinks(occurrence, ["SSHKey"], false, false)]
+    [#local baselineAttributes = baselineLinks["SSHKey"].State.Attributes]
```

Comment:
Do we need ssh keys?

Reply:
@roleyfoley I discussed this with @ml019 this morning and that eventuated into the idea outlined in #124 . If you agree with this approach I suggest we proceed with this  as-is  (considering the same approach is used elsewhere) and we'll refactor them all together?

Reply:
Yep #124 sounds good and we should put accross all providers too